acstools/polarization_tools.py

```diff
@@ -1,85 +1,328 @@
+
+        self.data = input_dict
 
-        self.wfc_transmission = Table(self.table_data['transmission']['wfc'],
+        self.wfc_transmission = Table(self.data['transmission']['wfc'],
                                       names=('filter', 't_para', 't_perp', 'correction'))
 
-        self.hrc_transmission = Table(self.table_data['transmission']['hrc'],
+        self.hrc_transmission = Table(self.data['transmission']['hrc'],
                                       names=('filter', 't_para', 't_perp', 'correction'))
 
-        self.wfc_efficiency = Table(self.table_data['efficiency']['wfc'],
-                                    names=('filter', 'pol0', 'pol60', 'pol120'))
+        self.wfc_efficiency = Table(self.data['efficiency']['wfc'], names=('filter', 'pol0', 'pol60', 'pol120'))
 
-        self.hrc_efficiency = Table(self.table_data['efficiency']['hrc'],
-                                    names=('filter', 'pol0', 'pol60', 'pol120'))
+        self.hrc_efficiency = Table(self.data['efficiency']['hrc'], names=('filter', 'pol0', 'pol60', 'pol120'))
+
+    @classmethod
+    def from_yaml(cls, yaml_file):
+        """
+        Read in a YAML file containing polarizer calibration data.
+
+        Parameters
+        ----------
+        yaml_file : str
+            Path to the YAML file containing the polarizer calibration information.
+
+        Returns
+        -------
+        acstools.polarizer_tools.PolarizerTables object
+        """
+        with open(yaml_file, 'r') as yf:
+            input_dict = yaml.safe_load(yf)
+        return cls(input_dict)
+
+    @classmethod
+    def from_package_data(cls):
+        """
+        Use the YAML file contained within the acstools package to retrieve the polarizer
+        calibration data.
+
+        Returns
+        -------
+        acstools.polarizer_tools.PolarizerTables object
+        """
+        filename = get_pkg_data_filename(os.path.join('data', 'polarizer_tables.yaml'))
+        return cls.from_yaml(filename)
 
 
 class Polarization:
     """
-    Class for handling ACS polarization data.
+    Class for handling ACS polarization data. Input data for this class come from
+    photometry of ACS polarization images. The methods associated with this class
+    will transform the photometry into Stokes parameters and polarization properties,
+    i.e., the polarization fraction and position angle.
+
+    Parameters
+    ----------
+    pol0 : float
+        Photometric measurement in POL0 filter. Units: electrons or electrons/second.
+    pol60 : float
+        Photometric measurement in POL60 filter. Units: electrons or electrons/second.
+    pol120 : float
+        Photometric measurement in POL120 filter. Units: electrons or electrons/second.
+    filter_name : str
+        Name of the filter crossed with the polarization filter, e.g., F606W.
+    detector : ['wfc', 'hrc']
+        Name of the ACS detector used for the observation. Must be either WFC or HRC.
+    pav3 : float
```

Comment:
Quantity?

Reply:
`quantity_input` doesn't quite work the way I'd like (it only checks if something is a quantity object and if not throws an exception), so I'm going to sneakily accept a quantity here but strip off the units (after ensuring it's in degrees) and then return a quantity at the end. The math in the middle just gets messy with having to throw in units on everything because it won't let me nicely add scalars to quantities.